packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1120,4 +1138,50 @@ apiDescribe('Database', (persistence: boolean) => {
       }).to.throw();
     });
   });
+
+  it('can wait for pending writes as expected', async () => {
+    await withTestDoc(persistence, async docRef => {
+      const firestore = docRef.firestore;
+      // Prevent pending writes receiving acknowledgement.
+      await firestore.disableNetwork();
+
+      const awaitPendingWrites1 = waitForPendingWrites(firestore);
+      const pendingWrites = docRef.set({ foo: 'bar' });
+      const awaitPendingWrites2 = waitForPendingWrites(firestore);
+
+      // `awaitsPendingWrites1` resolves immediately because there are no pending writes at
+      // the time it is created.
+      await expect(awaitPendingWrites1).to.be.eventually.fulfilled;
+
+      // pending writes can receive acknowledgements now.
+      await firestore.enableNetwork();
+      await expect(pendingWrites).to.be.eventually.fulfilled;
```

Comment:
`await expect(pendingWrites).to.be.eventually.fulfilled` is the same as `await pendingWrites`.

Reply:
Done.